dfusion_rust_core/src/models/solution.rs

```diff
@@ -29,24 +30,37 @@ impl Solution {
 
 impl Serializable for Solution {
     fn bytes(&self) -> Vec<u8> {
+        let mut res = (self.prices.len() as u16).to_be_bytes().to_vec();
         let alternating_buy_sell_amounts: Vec<u128> = self
             .executed_buy_amounts
             .iter()
             .zip(self.executed_sell_amounts.iter())
             .flat_map(|tup| once(tup.0).chain(once(tup.1)))
             .cloned()
             .collect();
-        [&self.prices, &alternating_buy_sell_amounts]
+
+        let prices_and_volumes: Vec<u8> = [&self.prices, &alternating_buy_sell_amounts]
             .iter()
             .flat_map(|list| list.iter())
             .flat_map(Serializable::bytes)
-            .collect()
+            .collect();
+        res.extend(prices_and_volumes);
+        return res;
     }
 }
 
 impl Deserializable for Solution {
     fn from_bytes(mut bytes: Vec<u8>) -> Self {
-        let volumes = bytes.split_off(TOKENS as usize * 12);
+        // First 2 bytes encode the length of price vector (i.e. num_tokens)
+        let len_prices = u16::from_be_bytes(
```

Comment:
nit: why not use split off here as well just like below? Also to avoid unwrapping, your could use a method like this one: https://github.com/gnosis/dex-services/blob/8473762420ceefac762799c823c659a212a1084d/dfusion_rust_core/src/models/util.rs#L207-L212

Reply:
The link is dead, but I will check it out.

Reply:
Ya, looks much cleaner than what I have here. I was just trying to "consume" the bytes instead of copy them (since they aren't needed anymore afterwards).